src/app/FakeLib/SemVerHelper.fs

```diff
@@ -76,6 +76,15 @@ type SemVerInfo =
                     0
             | _ -> invalidArg "yobj" "cannot compare values of different types"
 
+
+let private SemVerPattern = "^(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?(?:\+[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?$"
```

Comment:
You're saying we should throw an exception during `parse` for parsable but technically invalid semver strings? Int32.Parse doesn't align with the spec, particularly if we think about culture. While we can improve parse (and possibly make it more strict), some things (like prohibiting leading zeroes) are more complex without a regex.